### PR TITLE
chore: update tasklist comparator

### PR DESCRIPTION
**Pull request by hamidzr**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hamidzr** [comment]
let me know if you have any suggestions to better organize the code.

File: master/internal/resourcemanagers/priority.go [hunk 1]
```diff
@@ -343,21 +343,19 @@ func taskFilter(label string, zeroSlots bool) func(*sproto.AllocateRequest) bool
...
+	switch aReqComparator(a, b) {
+	case 0:
 		return false
+	case 1:
+		return false
+	case -1:
+		return true
+	default:
+		panic("unexpected comparison result")
 	}
```
Comment (stoksc) [inline master/internal/resourcemanagers/priority.go:358]: this, and maybe we dont need this func
```suggestion
	return aReqComparator(a, b) < 0
```

**Review by stoksc — APPROVED** [review]
lgtm, modulo comments

File: master/internal/resourcemanagers/priority.go (hunk 1 above)
Comment (hamidzr) [reply master/internal/resourcemanagers/priority.go:358]: duh lol

Comment (hamidzr) [reply master/internal/resourcemanagers/priority.go]: I removed it alltogether

File: master/internal/resourcemanagers/task_list.go [hunk 2]
```diff
@@ -112,3 +116,26 @@ func taskComparator(a interface{}, b interface{}) int {
 	}
 	return strings.Compare(string(t1.AllocationID), string(t2.AllocationID))
 }
+
+// aReqComparator compares AllocateRequests by how long their jobs have been submitted
+// while falling back to when their Allocation actor was created for non-job tasks.
+// a < b iff a is older than b.
+// The result will be 0 if a==b, -1 if a < b, and +1 if a > b.
+func aReqComparator(a *sproto.AllocateRequest, b *sproto.AllocateRequest) int {
```
Comment (stoksc) [inline master/internal/resourcemanagers/task_list.go:124]: i would rename this, personally. but most good names are long.

**Pull request merged**